File: tony/java/SimplifyPath.java
public class Solution {
    public String simplifyPath(String path) {
        String ret = "";
        String[] strs = path.split("/+");
        
        for (int i = 1; i < strs.length; i++) {
            if (strs[i].equals(".")) {
                continue;
            } else if (strs[i].equals("..")) {
                if (!ret.contains("/"))
                    ret += "/";
                ret = ret.substring(0, ret.lastIndexOf("/"));
            } else {
                ret += "/" + strs[i];
            }
        }
        
        return ret.equals("") ? "/" : ret;
    }
}

File: lulu/simplifyPath.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        aStack = []
        for x in path.strip().split('/'):
            if x == '..':
                if aStack:
                    aStack.pop()
            elif not x or x == '.':
                pass
            else:
                aStack.append(x)
        canical_path = '/'.join(aStack)
        return '/' + canical_path
